src/Microsoft.OpenApi.Hidi/Program.cs

```diff
@@ -127,5 +128,29 @@ static async Task Main(string[] args)
             //// Wait for logger to write messages to the console before exiting
             await Task.Delay(10);
         }
+
+        /// <summary>
+        /// This event is raised when the user presses either of the two breaking key combinations: Ctrl+C or Ctrl+Break keys.
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="eventArgs"></param>
+        private static void Console_CancelKeyPressEvent(object sender, ConsoleCancelEventArgs eventArgs)
+        {
+            if ((eventArgs.SpecialKey == ConsoleSpecialKey.ControlC) || (eventArgs.SpecialKey == ConsoleSpecialKey.ControlBreak))
+            {
+                Console.WriteLine("CTRL+C pressed, aborting current process...");
+                Thread.Sleep(5000);
```

Comment:
won't that make the process take always 5 seconds to exit no matter what?

Reply:
Yeah I thought adding a bit of delay would enhance the user experience. I can remove this..